x/storage/keeper/cross_app_permission.go

```diff
@@ -0,0 +1,222 @@
+	}
+
+	var operationType uint8
+	var result sdk.ExecuteResult
+	switch p := pack.(type) {
+	case *types.CreatePolicySynPackage:
+		operationType = types.OperationCreatePolicy
+		result = app.handleCreatePolicyFailAckPackage(ctx, appCtx, p)
+	case *types.DeletePolicySynPackage:
+		operationType = types.OperationDeletePolicy
+		result = app.handleDeletePolicyFailAckPackage(ctx, appCtx, p)
+	default:
+		panic("unknown cross chain ack package type")
+	}
+
+	if len(result.Payload) != 0 {
+		wrapPayload := types.CrossChainPackage{
+			OperationType: operationType,
+			Package:       result.Payload,
+		}
+		result.Payload = wrapPayload.MustSerialize()
+	}
+
+	return result
+}
+
+func (app *PermissionApp) ExecuteSynPackage(ctx sdk.Context, appCtx *sdk.CrossChainAppContext, payload []byte) sdk.ExecuteResult {
+	pack, err := types.DeserializeCrossChainPackage(payload, types.PermissionChannelId, sdk.SynCrossChainPackageType)
+	if err != nil {
+		panic("deserialize Policy cross chain package error")
+	}
+
+	var operationType uint8
+	var result sdk.ExecuteResult
+	switch p := pack.(type) {
+	case *types.CreatePolicySynPackage:
+		operationType = types.OperationCreatePolicy
+		result = app.handleCreatePolicySynPackage(ctx, p)
+	case *types.DeletePolicySynPackage:
+		operationType = types.OperationDeletePolicy
+		result = app.handleDeletePolicySynPackage(ctx, p)
+	default:
+		return sdk.ExecuteResult{
+			Err: types.ErrInvalidCrossChainPackage,
+		}
+	}
+
+	if len(result.Payload) != 0 {
+		wrapPayload := types.CrossChainPackage{
+			OperationType: operationType,
+			Package:       result.Payload,
+		}
+		result.Payload = wrapPayload.MustSerialize()
+	}
+
+	return result
+}
+
+func (app *PermissionApp) handleDeletePolicyAckPackage(ctx sdk.Context, appCtx *sdk.CrossChainAppContext, ackPackage *types.DeletePolicyAckPackage) sdk.ExecuteResult {
+	return sdk.ExecuteResult{}
+}
+
+func (app *PermissionApp) handleDeletePolicyFailAckPackage(ctx sdk.Context, appCtx *sdk.CrossChainAppContext, synPackage *types.DeletePolicySynPackage) sdk.ExecuteResult {
+	return sdk.ExecuteResult{}
+}
+
+func (app *PermissionApp) handleDeletePolicySynPackage(ctx sdk.Context, deletePolicyPackage *types.DeletePolicySynPackage) sdk.ExecuteResult {
+	err := deletePolicyPackage.ValidateBasic()
+	if err != nil {
+		return sdk.ExecuteResult{
+			Payload: types.DeletePolicyAckPackage{
+				Status:    types.StatusFail,
+				ExtraData: deletePolicyPackage.ExtraData,
+			}.MustSerialize(),
+			Err: err,
+		}
+	}
+
+	policy, found := app.permissionKeeper.GetPolicyByID(ctx, math.NewUintFromBigInt(deletePolicyPackage.Id))
+	if !found {
+		return sdk.ExecuteResult{
+			Payload: types.DeletePolicyAckPackage{
+				Status:    types.StatusFail,
+				ExtraData: deletePolicyPackage.ExtraData,
+			}.MustSerialize(),
+			Err: types.ErrNoSuchPolicy,
+		}
+	}
+
+	_, err = app.permissionKeeper.DeletePolicy(ctx, policy.Principal, policy.ResourceType, policy.ResourceId)
```

Comment:
We need verify whether the `operator` is allowed to delete the policy.

Reply:
added check